### PR TITLE
Fixes issue #5625

### PR DESCRIPTION
**Pull request by zachrab**
@jhass Please let me know if this is all that's needed for this fix.

### CODE REVIEW

**Comment by jhass** [comment]
Okay, please do two more things please:

Add a spec [here](https://github.com/diaspora/diaspora/blob/develop/spec/models/user_spec.rb#L947)  that starts with a user which has the field set to false and asserts that after calling the method, it's set to true. Maybe also backfill a test for the fields that are set to false there.

Add a migration that sets it to `true` for all already closed accounts. See [here](https://github.com/diaspora/diaspora/blob/develop/app/presenters/statistics_presenter.rb#L50) for how to select these (the blank username is indicating an old invite, not a closed account).

**Comment by zachrab** [comment]
@jhass What do you mean by "backfill a test for the fields that are set to false there"?

File: app/models/user.rb [hunk 1]
```diff
@@ -485,10 +485,10 @@ def clear_account!
       self[field] = nil
     end
     [:getting_started,
-     :disable_mail,
      :show_community_spotlight_in_stream].each do |field|
```
Comment (jhass) [inline app/models/user.rb]: I can't seem to spot an existing test for these fields, that's what I mean with backfilling ;)

File: db/migrate/20150209230946_disable_mail_for_closed_account.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class DisableMailForClosedAccount < ActiveRecord::Migration
+  def up
+  	closed_users = User.joins(:person).where(people: {closed_account: true})
+  	closed_users.each do |closed_user|
+  		closed_user.update_attribute(:disable_mail, true)
+  	end
```
Comment (jhass) [inline db/migrate/20150209230946_disable_mail_for_closed_account.rb]: Let's use [update_all](http://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-update_all) on the relation here, that should be much faster.

**Comment by jhass** [comment]
Please update `db/schema.rb` to the new migration. Since the migration contains no schema changes, you shouldn't commit them if you have any. I also think it's small enough to be rebased into a single commit:

```bash
git remote add upstream git://github.com/diaspora/diaspora.git
git fetch upstream
git checkout 5625-disable-mail-to-deleted-user
git rebase -i upstream/develop
# Leave the first line at pick, choose squash for the others
bin/rake db:migrate
git commit --amend -p db/schema.rb
git push -f origin 5625-disable-mail-to-deleted-user
```

The code itself looking good.

**Comment by jhass** [comment]
@zachrab ping, still interested in getting this merged ;)

**Comment by zachrab** [comment]
@jhass sorry for the delay!

**Comment by jhass** [comment]
Thank you!

**Pull request merged**